sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/accessibility/AriaProptypesCheck.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * SonarSource HTML analyzer :: Sonar Plugin
+ * Copyright (c) 2010-2024 SonarSource SA and Matthijs Galesloot
+ * sonarqube@googlegroups.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sonar.plugins.html.checks.accessibility;
+
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.sonar.check.Rule;
+import org.sonar.plugins.html.checks.AbstractPageCheck;
+import org.sonar.plugins.html.checks.accessibility.Aria.AriaProperty;
+import org.sonar.plugins.html.checks.accessibility.Aria.AriaPropertyType;
+import org.sonar.plugins.html.node.TagNode;
+
+import static org.sonar.plugins.html.checks.accessibility.Aria.ARIA_PROPERTIES;
+
+@Rule(key = "S6793")
+public class AriaProptypesCheck extends AbstractPageCheck {
+
+  @Override
+  public void startElement(TagNode element) {
+    var attributes = element.getAttributes();
+    for (var attribute : attributes) {
+      var name = attribute.getName();
+      var normalizedName = name.toLowerCase(Locale.ENGLISH);
+
+      if (!ARIA_PROPERTIES.containsKey(normalizedName)) {
+        continue;
+      }
+
+      var value = attribute.getValue();
+      var property = ARIA_PROPERTIES.get(normalizedName);
+      if (!isValid(property, value)) {
+        createViolation(element.getStartLinePosition(), message(name, property.getType(), property.getValues()));
+      }
+    }
+  }
+
+  private static boolean isValid(AriaProperty property, String value) {
```

Comment:
These might be candidates to be moved to the `Aria.java` helper class.

Reply:
I don't think we need to. This is really specific to the rule, and the JavaScript implementation doesn't expose it as a helper function anywhere outside the rule.